src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -443,6 +444,8 @@ public Response get(@Context HttpHeaders httpHeaders, String json) throws Except
 
 		try (Scope scope = tracer.scopeManager().activate(span, false))
 		{
+			final long start = System.currentTimeMillis();
```

Comment:
hmm, is there really no way to detect if the client disconnected? that would be the cleaner approach IMO

Reply:
ideally, in a microservice architecture, the timeouts are passed on from the client and the server uses a chunk of that timeout and forwards it to the next server in the flow. In our case the controller does not pass the timeout to kairos, So we have to set it explicitly at kairos level and check it and not based on client disconnects

Reply:
which means e.g. if the controller is restarted for some reason the request still continues for no reason until this timeout is hit

Reply:
This is already the behavior as it is not possible to know bout a client disconnect until we try to write to it.  This change will at least reduce the maximum amount of time Kairos will spin on a request that no longer has a receiver.  We catch IOException below in that case.

Reply:
Actually, diving deeper into the internals of how the KairosDatastore is currently implemented, each call made to it from the resource level is in itself an indeterminate number of calls to its underlying C* datastore.  This approach will not work, as checking per iteration of the outer call could just as easily hang, or greatly exceed the timeout, on account of one of the very many inner calls.  We will need to use an async executor with a timeout mechanism, and wait on that.  I'll followup with an example shortly.